crypto/ec/ecp_s390x_nistp.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include "ec_lcl.h"
+#include "s390x_arch.h"
+
+/* Size of parameter blocks */
+#define S390X_SIZE_PARAM		4096
+
+/* Size of fields in parameter blocks */
+#define S390X_SIZE_P256			32
+#define S390X_SIZE_P384			48
+#define S390X_SIZE_P521			80
+
+/* Offsets of fields in PCC parameter blocks */
+#define S390X_OFF_RES_X(n)		(0 * n)
+#define S390X_OFF_RES_Y(n)		(1 * n)
+#define S390X_OFF_SRC_X(n)		(2 * n)
+#define S390X_OFF_SRC_Y(n)		(3 * n)
+#define S390X_OFF_SCALAR(n)		(4 * n)
+
+/* Offsets of fields in KDSA parameter blocks */
+#define S390X_OFF_R(n)			(0 * n)
+#define S390X_OFF_S(n)			(1 * n)
+#define S390X_OFF_H(n)			(2 * n)
+#define S390X_OFF_K(n)			(3 * n)
+#define S390X_OFF_X(n)			(3 * n)
+#define S390X_OFF_RN(n)			(4 * n)
+#define S390X_OFF_Y(n)			(4 * n)
```

Comment:
Does this have the same offset as the field above?

Reply:
Yes its two different param blocks for sign and verify. sign is R-S-hash-privkey-random, verify is R-S-hash-pubX-pubY. Structures for each param block would be more readable but since member size is different for each curve, i chose to go with offsets, otherwise whole file would be macros parametrized by member size,